pkg/network/vmicontroller/status.go

```diff
@@ -54,17 +63,14 @@ func UpdateStatus(vmi *v1.VirtualMachineInstance, pod *k8scorev1.Pod) error {
 func calculatePrimaryIfaceStatus(
 	vmi *v1.VirtualMachineInstance,
 	networkStatuses []networkv1.NetworkStatus,
+	clusterConfig clusterConfigChecker,
 ) []v1.VirtualMachineInstanceNetworkInterface {
 	primaryNetworkSpec := vmispec.LookUpDefaultNetwork(vmi.Spec.Networks)
 	if primaryNetworkSpec == nil {
 		return nil
 	}
 
-	primaryPodIfaceName := multus.LookupPodPrimaryIfaceName(networkStatuses)
-	if primaryPodIfaceName == "" {
-		primaryPodIfaceName = namescheme.PrimaryPodInterfaceName
-	}
-
+	primaryPodIfaceName := calculatePrimaryPodInterfaceName(networkStatuses, clusterConfig)
```

Comment:
I do not really understand why we need this extra functions.

How about this (adding 3 lines):
```go
primaryPodIfaceName := multus.LookupPodPrimaryIfaceName(networkStatuses)
if primaryPodIfaceName == "" {
    primaryPodIfaceName = namescheme.PrimaryPodInterfaceName
}

if !clusterConfig.DynamicPodInterfaceNamingEnabled() {
    primaryPodIfaceName = ""
}
```

Originally, we have not treated the primary interface, and I think we should keep that protected by the FG.
That means we should have probably done something like this at a level up:
```go
if !clusterConfig.DynamicPodInterfaceNamingEnabled() {
    interfaceStatuses = append(interfaceStatuses, calculatePrimaryIfaceStatus(vmi, networkStatuses, clusterConfig)...)
}
```

Reply:
Sorry, I meant `if clusterConfig.DynamicPodInterfaceNamingEnabled()` not with a `!`.